templates/batch-llm/util/utils.py

```diff
@@ -36,3 +41,35 @@ def get_a10g_or_equivalent_accelerator_type() -> str:
     instead.
     """
     return "L4" if _on_gcp_cloud() else "A10G"
+
+
+def prompt_for_hugging_face_token(hf_model: str) -> str:
+    """Prompts the user for Hugging Face token if required by the model.
+    Returns the token as a string. If a token is not required by the model,
+    returns an empty string."""
+
+    try:
+        # Try loading model config to check if Hugging Face token is required.
+        AutoConfig.from_pretrained(hf_model)
+        # If config is loaded successfully, no need for token.
+        return ""
+    except OSError:
```

Comment:
Are there other OSErrors that could come from the above? Should we check the message of the exception to make sure it matches?

Reply:
good point, added a check for the specific error type